### PR TITLE
Math and currency symbols are safe for display

### PR DESCRIPTION
**Pull request by DemiMarie**
This also cleans up some technical debt: a Python script generated C code, which was in turn used to generate more C code!  Instead, just use ICU4C in the C code directly, deleting one of two code generators.

### CODE REVIEW

**Comment by marmarek** [comment]
> Math and currency symbols are safe for display

Some may be safe to display, but some may still try to exploit bugs in some tools/scripts processing file names. I'd prefer to refuse by default shell meta-characters at the very least. Do you think it's too strict?

**Comment by marmarek** [comment]
Maybe it is too strict, because such buggy tools would trip already on spaces. So, maybe it's okay?

**Comment by DemiMarie** [comment]
> > Math and currency symbols are safe for display
> 
> Some may be safe to display, but some may still try to exploit bugs in some tools/scripts processing file names. I'd prefer to refuse by default shell meta-characters at the very least. Do you think it's too strict?

I’m not sure, but that would be a separate change.

**Comment by marmarek** [comment]
PipelineRetry

File: qrexec-lib/unicode-allowlist-table.c [hunk 1]
```diff
@@ -1,12 +1,12 @@
-    case 0x000020 ... 0x00007e:
-    case 0x0000a1:
+    case 0x000020 ... 0x00005d:
+    case 0x00005f:
+    case 0x000061 ... 0x00007e:
```
Comment (marmarek) [inline qrexec-lib/unicode-allowlist-table.c]: So, now "^" and "`" are excluded, is that intentional? The latter might be a good idea, but the commit message talks only about allowing some more...

Comment (DemiMarie) [reply qrexec-lib/unicode-allowlist-table.c]: Nope, that’s a bug.  Maybe I should create a file with every allowed character and then add a test to make sure they stay allowed!

**Review by marmarek — APPROVED** [review]

**Pull request merged**